classes/resin-u-boot: Increase OS_BOOTCOUNT_LIMIT to 3

acostach opened the pull request:
Let's increase the current value for the rollback-altboot
bootcount to 5, to account for accidental poweroffs triggered
by users or by power supplys when booting into the new OS,
during HUP.

When the limit is reached, u-boot will proceed to try
to boot into the old OS.

Change-type: patch
Signed-off-by: Alexandru Costache <alexandru@balena.io>

Connects-to: https://github.com/balena-os/meta-balena/issues/2301
---
### Contributor checklist
<!-- For completed items, change [ ] to [x].  -->
- [ ] Changes have been tested
  - [ ] Covered in automated test suite
  - [ ] Manual test case recorded
- [x] `Change-type` present on at least one commit
- [x] `Signed-off-by` is present
- [x] The PR complies with the [Open Embedded Commit Patch Message Guidelines](http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)
<!-- optional: `Changelog-entry` present on at least one commit if you want to set the changelog entry manually-->

### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

acostach commented:
@resin-jenkins test this please

alexgg reviewed (changes requested):
lgtm. If we are thinking about accidental poweroffs, 5 seems too many attempts, I'd consider 3 not to delay the rollback too much in case of a real problem.
Feel free to self-certify anyway.

acostach commented:
> lgtm. If we are thinking about accidental poweroffs, 5 seems too many attempts, I'd consider 3 not to delay the rollback too much in case of a real problem.
> Feel free to self-certify anyway.

Ack, thanks Alex, I updated the commit and switched to 3 attempts

acostach commented:
@balena-ci I self-certify!

acostach commented:
@balena-ci rebase

acostach commented:
@balena-ci rebase

acostach commented:
@alexgg looks like this will might an explicit approval for the requested changes, can't merge with self-certify unfortunately. Thank you

jakogut commented:
@alexgg @acostach This is a known problem with GitHub. If a reviewer requests changes, another reviewer (such as versionbot during self-certification) cannot override the reviewer that requested changes. Either the original reviewer must give approval, or the PR must be closed, a new PR opened, and the self-certify redone.

jakogut reviewed (approved):
Looks good, though I think we should also look into continuous fallback support.

alexgg commented:
@balena-ci rebase

alexgg reviewed (approved):
lgtm

klutchell commented:
@balena-ci rebase

klutchell commented:
@balena-ci rebase

alexgg commented:
@resin-jenkins retest this please

acostach commented:
@balena-ci rebase

alexgg commented:
Addressed from v2.85.10
